### PR TITLE
updateConnection

### PR DESCRIPTION
**Pull request by eprince1**
connect to project in account

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -64,29 +76,40 @@ async function connect() {
...
+  dbRef.onDisconnect(() => connect());
+}
+
 async function getEndpoint(projectId) {
-  const endpointId = await axios.get(
-    `https://us-central1-jellysync.cloudfunctions.net/api/projectEndpoint?projectId=${projectId}`
-  );
-  localStorage.setItem('jellysyncEndpoint', endpointId.data);
+  const prdUrl = 'https://us-central1-jellysync.cloudfunctions.net/api';
+  const localUrl = 'http://localhost:5001/jellysync/us-central1/api';
```
Comment (avisprince) [inline index.js]: what are we doing about these? Do we need a set-env?

Comment (eprince1) [reply index.js]: o yeah, forget about this

File: index.js [hunk 2]
```diff
@@ -51,11 +57,17 @@ async function connect() {
   }
 
   try {
-    dbRef.on('value', snapshot => {
+    dbRef.on('value', async snapshot => {
       // successful connection resets attempts
+      console.log('connected');
```
Comment (avisprince) [inline index.js]: is this needed?

Comment (eprince1) [reply index.js]: nope

File: index.js [hunk 3]
```diff
@@ -15,12 +15,23 @@ let endpoint = null;
...
 let attempts = 4;
+let initialLoad = true;
+let database = null;
+
+async function initialize(projectId) {
+  endpoint = localStorage.getItem('jellysyncEndpoint');
+
+  if (!endpoint) {
+    endpoint = await getEndpoint(projectId);
+    console.log(endpoint);
```
Comment (avisprince) [inline index.js]: another one

File: index.js [hunk 4]
```diff
@@ -30,16 +41,11 @@ async function initialize(projectId, options) {
...
 
   currVersion = localStorage.getItem('jellySyncVersion');
-  endpoint = localStorage.getItem('jellysyncEndpoint');
-
-  if (!endpoint) {
-    endpoint = await getEndpoint(projectId);
-  }
 
-  dbRef = database.ref(`project/${endpoint}`);
+  dbRef = database.ref(`/projects/${projectId}/${endpoint.id}`);
```
Comment (avisprince) [inline index.js:47]: isnt it possible for endpoint to be undefined here?

Comment (eprince1) [reply index.js:47]: what should we do in that case?

Comment (avisprince) [reply index.js:47]: 1. we should put retry logic into axios (npm axios-retry)
2. for now check for an empty value and bail gracefully

**Pull request merged**